behavior/datagen/gen.py

```diff
@@ -537,23 +535,54 @@ def main(self):
             # Create the output directory.
             mode_dir = Path(self.dir_output) / mode / experiment_name
             Path(mode_dir).mkdir(parents=True)
-            # Copy the config file to the output directory.
+            # Copy the configuration file to the output directory.
             shutil.copy(config_path, mode_dir)
 
+            # Build sweeping space.
+            ps_space = param_sweep_space(self.config["param_sweep"])
             # For each benchmark, ...
             for benchmark in benchmarks:
-                # Run BenchBase to generate training data.
-                results_dir = Path(mode_dir / benchmark)
-                results_dir.mkdir()
-                benchbase_results_dir = Path(results_dir / "benchbase")
-                benchbase_results_dir.mkdir(exist_ok=True)
-                logger.info(
-                    "Running experiment %s with benchmark %s and results %s",
-                    experiment_name,
-                    benchmark,
-                    results_dir,
-                )
-                self.run_experiment(benchmark, results_dir, benchbase_results_dir)
+                input_cfg_path = self.dir_benchbase_config / f"{benchmark}_config.xml"
+
+                def sweep_func(parameters, closure):
+                    """Callback to datagen parameter sweep.
+
+                    Parameters:
+                    -----------
+                    parameters: List[Tuple(List[str], Any)]
```

Comment:
```suggestion
                    parameters: List[Tuple[List[str], Any]]
```

(it is annoying and kinda ugly, but Python type annotations are all `[]` to my knowledge)